File: CWM.Services/Models/Vehicle.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CWM.Database.Models
{
    [Table("Vehicles")]
    public class Vehicle
    {
        [Key]
        public int Id { get; set; }

        public string Chassis {  get; set; } = string.Empty;
        public string Brand { get; set; } = string.Empty;
        public string Model { get; set; } = string.Empty;
        public int CubicCapacity { get; set; }
        public int Kilowatts { get; set; }
        public string Transmision { get; set; } = string.Empty;
        public DateTime ProductionDate { get; set; }
        public string Fuel { get; set; } = string.Empty;

        [ForeignKey("User")]
        public int UserId { get; set; }
        public virtual User? User { get; set; }

        [ForeignKey("VehicleServiceHistory")]
        public int? VehicleServiceHistoryId { get; set; }
        public virtual VehicleServiceHistory? ServiceHistory { get; set; }


        // Relations
        public virtual ICollection<WorkOrder> WorkOrders { get; set; } = new List<WorkOrder>();
    }
}

File: CWM.UI/cwm_desktop_mobile/lib/models/vehicle.dart
import 'package:cwm_desktop_mobile/models/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'vehicle.g.dart';

@JsonSerializable()
class Vehicle {
  int id;
  String chassis;
  String brand;
  String model;
  int cubicCapacity;
  int kilowatts;
  String transmision;
  DateTime productionDate;
  String fuel;
  User? user;

  Vehicle(
      this.id,
      this.chassis,
      this.brand,
      this.model,
      this.cubicCapacity,
      this.kilowatts,
      this.transmision,
      this.productionDate,
      this.fuel,
      this.user);

  factory Vehicle.fromJson(Map<String, dynamic> json) =>
      _$VehicleFromJson(json);

  Map<String, dynamic> toJson() => _$VehicleToJson(this);
}
